### ISSUE TITLE
Many ConfigurationBindingGeneratorTests fails on CI

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/runtime#107381]
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=797859
Build error leg or test failing: Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests.ConfigurationBindingGeneratorTests.Bind_Key_Instance
Pull request: https://github.com/dotnet/runtime/pull/106781

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": "System.OutOfMemoryException : Insufficient memory to continue the execution of the program.",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

**Comment by tarekgh** [comment]
This is the first time I have seen such failure. My guess is the running machine was in a bad state which couldn't even allocate native memory. We'll watch if we see the failure again soon. 

@buyaa-n did you try to re-run this failed test in PR after you saw it failed? or you just logged the issue?

CC @ericstj

**Comment by tarekgh** [comment]
Seeing the same failure in other PRs https://github.com/dotnet/runtime/pull/107009.

**Comment by buyaa-n** [comment]
> @buyaa-n did you try to re-run this failed test in PR after you saw it failed?

Tried rerun in another PR, did not help

**Comment by buyaa-n** [comment]
> Actually in another PR, rerun made the test pass. will watch that and find out what is going on.

Ah, for me it just added to the known build errors instead succeed.

**Comment by ericstj** [comment]
Such a high recent hit count does signal a real problem here.  We should have a look and see if we can bisect what caused the regression.

**Comment by tarekgh** [comment]
I have done `git bisect` and isolated the commit that started causing the regression. @elinor-fung could you please have a look? Looks your change started to cause this OOM problem (specifically on x86). Thanks!

```
commit 815509cb9dba9fe72adc11119c2336e916a92966
Author: Elinor Fung <elfung@microsoft.com>
Date:   Wed Sep 4 07:23:26 2024 -0700

    Put handling of sized ref handles under FEATURE_SIZED_REF_HANDLES (#107326)

 src/coreclr/debug/daccess/dacdbiimpl.cpp |  7 -------
 src/coreclr/debug/daccess/dacimpl.h      |  2 +-
 src/coreclr/debug/daccess/request.cpp    |  3 +--
 src/coreclr/gc/gc.cpp                    | 13 +++++++++++--
 src/coreclr/gc/gcinterface.h             |  2 ++
 src/coreclr/gc/gcscan.cpp                |  2 ++
 src/coreclr/gc/gcscan.h                  |  3 ++-
 src/coreclr/gc/handletableconstants.h    |  1 +
 src/coreclr/gc/objecthandle.cpp          | 25 ++++++++++++++++++++++---
 src/coreclr/vm/gchandleutilities.h       | 17 -----------------
 src/coreclr/vm/marshalnative.cpp         |  2 +-
 src/coreclr/vm/runtimehandles.cpp        |  2 +-
 12 files changed, 44 insertions(+), 35 deletions(-)
```

**Comment by mangod9** [comment]
@tarekgh, assume this in main and not in 9 branches?

**Comment by tarekgh** [comment]
> assume this in main and not in 9 branches?

That is right. We are seeing the failure in main and not in 9 release branch.

**Comment by mangod9** [comment]
ok just making sure since it was moved to 9 milestone. @elinor-fung, please take a look and maybe revert since its causing many CI failures?
